lib/MusicBrainz/Server/WebService/Serializer/JSON/LD/Role/Genre.pm

```diff
@@ -0,0 +1,30 @@
+package MusicBrainz::Server::WebService::Serializer::JSON::LD::Role::Genre;
+use Moose::Role;
+use MusicBrainz::Server::WebService::Serializer::JSON::LD::Utils qw( list_or_single serialize_entity );
+
+around serialize => sub {
+    my ($orig, $self, $entity, $inc, $stash, $toplevel) = @_;
+    my $ret = $self->$orig($entity, $inc, $stash, $toplevel);
+
+    my $tags = $stash->store($entity)->{top_tags};
+    my @genres = map { $_->{tag}->genre } grep { $_->{tag}->genre } @{ $tags };
+
+    if (@genres) {
+      $ret->{genre} = list_or_single(map { DBDefs->CANONICAL_SERVER . '/genre/' . $_->gid } @genres);
+    }
```

Comment:
I'd reduce this to a single map instead of two maps + a grep
```suggestion
    my @genres = map {
        my $genre = $_->{tag}->genre;
        $genre ? (DBDefs->CANONICAL_SERVER . '/genre/' . $genre->gid) : ()
    } @$tags;

    if (@genres) {
      $ret->{genre} = list_or_single(@genres);
    }
```

Reply:
(Also, looks like the indentation inside the `if` is off, should be 4 spaces instead of two.)